File: core/src/main/java/com/sequenceiq/cloudbreak/converter/ContainerInfoToContainerConverter.java
package com.sequenceiq.cloudbreak.converter;


import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import com.sequenceiq.cloudbreak.domain.Container;
import com.sequenceiq.cloudbreak.orchestrator.model.ContainerInfo;

@Component
public class ContainerInfoToContainerConverter extends AbstractConversionServiceAwareConverter<ContainerInfo, Container> {
    private static final Logger LOGGER = LoggerFactory.getLogger(ContainerInfoToContainerConverter.class);

    @Override
    public Container convert(ContainerInfo source) {
        Container container = new Container();
        container.setContainerId(source.getId());
        container.setName(source.getName());
        container.setImage(source.getImage());
        container.setHost(source.getHost());
        return container;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/ContainerInfoToContainerConverter.kt
package com.sequenceiq.cloudbreak.converter


import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.domain.Container
import com.sequenceiq.cloudbreak.orchestrator.model.ContainerInfo

@Component
class ContainerInfoToContainerConverter : AbstractConversionServiceAwareConverter<ContainerInfo, Container>() {

    override fun convert(source: ContainerInfo): Container {
        val container = Container()
        container.containerId = source.id
        container.name = source.name
        container.image = source.image
        container.host = source.host
        return container
    }

    companion object {
        private val LOGGER = LoggerFactory.getLogger(ContainerInfoToContainerConverter::class.java)
    }
}
